### PR TITLE
[READY] Audio refactors, fixes, and general dumbshittery

### PR DESCRIPTION
**Pull request by actioninja**
The 543 file diff with 1600 line changes was mostly regexing, I'm not that insane.

## About The Pull Request

Huge explanation because I did way too much in one PR!

### playsound has been refactored a lot:
- vary now does something different, it acts as a range of variation instead of a toggle. Things may sound a little different.
- frequency has been renamed to playback_rate and reordered, it's now a multiplier instead of a direct frequency call. Again, things may sound a little different.
- pitch variation works based on native byond playback rate changing instead of bullshit that assume a fixed sample rate and also fucked up its math
- distant sound support is now built directly into playsound instead of being snowflake code on meteors and explosions
### fixes were made:
- pierce.ogg which is used for a shitload of sound effects has been sped up to match how it sounded before the 44.1khz resample since it is literally used with old shitty fucked up pitch variation in the entire codebase causing it to playback much faster than the original sound file before it was resampled. tl;dr it sounds how it's supposed to, shooting people won't produce weird loud meaty thunks anymore.
- the grenade arming sound has been changed back to a beeping instead of the accidental saved over wrong sound, whoops
- pda and newscaster beeps playback at the correct pitch now
### refactors were done:
- Explosions and meteors now use playsound() parameters instead of a bigass for loop. They also both use a new helper function called shake_area() to handle screen shake. As a result, the exact behavior for both is different now, but is similar. **OH MY GOD EXPLOSION CODE CHANGES** just kidding it's only visual and audios,
- Shuttles no longer inexplicably play their sound effects from EVERY DOOR AT THE SAME TIME AT MAX VOLUME causing inconsistency in shuttle volume and massive earrape, instead playing the sound only once originating from the engine closest to you if you're close, or whatever the first engine in the engine list is for the distant souns. if there is no engine, there's no logical place for it to originate from, so it comes from doors again, but only the first door returned by a search for them. If there's no doors, it just plays without a source because fuck it

I think I did some other stuff but I don't remember.

## Why It's Good For The Game

Fixes, refactors, and disabled new features, oh my

## Changelog
:cl:
fix: fixed a couple of incorrect sounds and some sounds playing at the wrong pitch
soundadd: The shuttle can now be heard from further away with new distance sounds
refactor: playsound works a little differently on the backend now, especially in regards to pitch variation. Most sounds are likely to pitch vary a slight amount differently due to it working fundamentally differently. The function also has native support for distant sounds now.
/:cl:

I forgot to save the original regexes I used, so here's some very similar ones I wrote
These will work best used in this order
## REGEXES USED:
**ALL OF THESE ARE PCRE REGEXES, IF YOU'RE USING SED YOU'LL NEED TO USE THE EXTENDED FLAG**
### finds places where vary is true (or 1): 
`(playsound(_local)?\((\s?.+\s?,){3}\s?)(1|TRUE)(\s?(\)|.+\)))`
### replace command to replace with `DEFAULT_SOUND_VARY`:
`\1DEFAULT_SOUND_VARY\5`
### full command:
`/(playsound(_local)?\((\s?.+\s?,){3}\s?)(1|TRUE)(\s?(\)|.+\)))/\1DEFAULT_SOUND_VARY\5/g`
### finds places where vary is false (or 0), and then the function closes:
`(playsound(_local)?\((\s?.+\s?,){2}\s?.+\s?)(,\s?(0|FALSE)\s?)(\))`
### replace command to get rid of it because false is now the default:
`\1\6`
### full command:
`/(playsound(_local)?\((\s?.+\s?,){2}\s?.+\s?)(,\s?(0|FALSE)\s?)(\))/\1\6/g`

**`extrarange` was renamed to `extra_range` which should be caught by a simple find and replace.**
### here's a regex anyways:
`/extrarrange/extra_range/g`

**`frequency` was renamed to `playback_rate`, and repositioned, it's now after vary. frequency was a not very commonly used parameter so I just searched for `frequency` and then corrected it by hand when I did it.**
**Here's a regex that makes them easier to find at least, if your codebase for some reason extensively uses `frequency` you may find it worth it to write a proper replacement regex, but I didn't:**
`playsound.+frequency`

**The position of `extra_range` changed, and it was used as a direct, non-named call in a lot of places.**
### Here's a regex to find those places:
`(playsound\((\s?[^,]+\s?,){4}\s?)([^\n,]+\s?(\)\n|,.+\)))`
### A replacement to automatically named-parameterize those cases:
`\1extra_range = \3`
### full command:
`/(playsound\((\s?[^,]+\s?,){4}\s?)([^\n,]+\s?(\)\n|,.+\)))/\1extra_range = \3/g`

After using these, I recommend doing a simple search for `playsound` on whatever your repo's custom files are and trying to find edge cases (read: shitcode) that weren't covered by these, or places where they acted up.
I'm not very good with regex, and some wizard might be able to come and turn these into a spicy one liner, but this should at least work.

### CODE REVIEW

**Comment by actioninja** [comment]
Going to fix merge conflict and flatten commits tommorow it is late here

**Comment by zxaber** [comment]
>pda and newscaster beeps playback at the correct pitch now

Thanks for this.

**Comment by RaveRadbury** [comment]
A grateful station thanks you!

**Comment by Razharas** [comment]
I just hope the datum outputs framework is preserved

**Comment by actioninja** [comment]
@Razharas it is.

**Comment by Fox-McCloud** [comment]
`Rudimentary support for sound environments. They are not actually enabled.`

No, stop.

These don't work, and never have. Every time they're talked about and suggested, I have to pop up to remind people of why. Oh well.

Environments don't work because they apply globally to all current sounds and not singular instances of sounds. There's no way to really fix this and no way around it; it's just the way BYOND works.

If you play a sound with a sound environment, it'll apply that environment to every single sound played after setting the environment.

You can hack your way around it by reseting the environment back to what it originally was before a new sound plays, but this doesn't rectify the issue--it only masks it for short sounds.

If an existing sound is already playing (like music, or something that bypasses `playsound`) setting the environmental variable will change how the existing sound ...well, sounds.

This results in echoey music, adminhelps ringing out like it's a sewer, and weird oddities like an echo suddenly cutting off early because some other sound played and reset the environmental variable, so the echo just drops out completely.

IF you want to mess with environment, it has to be globally set and you kinda have to accept the consequences of it--Goon does this, for example; they have significant echo on station and just put up with that generally being applied to most sounds.

Otherwise, if you intend to have set areas or specific sounds have environment---just don't. It doesn't work like most people think it does.

**Comment by actioninja** [comment]
@Fox-McCloud If this is still accurate, that's pretty incredibly horrible. You'd think it would be per sound considering the environment is tied to the sound object, or at the very least per channel, but that makes too much sense.

I'll do some testing and if this is still true I'll strip out the environment stuff, it wasn't that much work.

**Comment by bobbahbrown** [comment]
can you post your regex for potential future use or reference

**Comment by actioninja** [comment]
@bobbahbrown I didn't save them, I should have somwhere. They weren't particularly complicated to write.

**Comment by AnturK** [comment]
Rebase so this can be testmerged please.

**Comment by actioninja** [comment]
I didn't realize there was conflicts because the merge conflict label isn't working right. Doing it now.

**Comment by actioninja** [comment]
man I fucked up git a lot trying to do that, it's good now though.

File: code/datums/outputs.dm [hunk 1]
```diff
@@ -25,10 +25,17 @@
...
+			var/high = playback_rate + vary
+			var/random_offset = (high - low) * rand()
+			playback_rate = low + random_offset
+
+		var/area/sound_area = get_area(turf_source)
+		if(sound_area && sound_area.sound_environment)
+			sound_output.environment = sound_area.sound_environment
+
+		if(receiver.override_sound_environment)
+			sound_output.environment = receiver.override_sound_environment
```
Comment (Fox-McCloud) [inline code/datums/outputs.dm]: You still have `environment` here.

Again, I've described the problem with how this functions.

Comment (actioninja) [reply code/datums/outputs.dm]: I actually just forgot to ever remove it.

Comment (Fox-McCloud) [reply code/datums/outputs.dm]: Great. Thank you!

**Comment by actioninja** [comment]
I just realized the second regexes I wrote were better than the first ones and found a shitload of cases the first ones missed.

**Comment by actioninja** [comment]
yeah fuck this pr it's fucked, gonna roll the smaller fixes into separate PRs and turn just the playsound refactor into its own.

**Pull request closed without merging**